charts/kube-prometheus-stack/hack/sync_grafana_dashboards.py

```diff
@@ -139,6 +139,34 @@ def new_representer(dumper, data):
 data:
 '''
 
+    # Add GrafanaDashboard custom resource
+grafana_dashboard_operator = """
+---
+{{- if and .Values.grafana.operator.dashboardsConfigMapRefEnabled (or .Values.grafana.enabled .Values.grafana.forceDeployDashboards) (semverCompare ">=%(min_kubernetes)s" $kubeTargetVersion) (semverCompare "<%(max_kubernetes)s" $kubeTargetVersion) .Values.grafana.defaultDashboardsEnabled%(condition)s }}
+apiVersion: grafana.integreatly.org/v1beta1
+kind: GrafanaDashboard
+metadata:
+  name: {{ printf "%%s-%%s" (include "kube-prometheus-stack.fullname" $) "%(name)s" | trunc 63 | trimSuffix "-" }}
+  namespace: {{ template "kube-prometheus-stack-grafana.namespace" . }}
+  labels:
+    {{- if $.Values.grafana.sidecar.dashboards.label }}
+    {{ $.Values.grafana.sidecar.dashboards.label }}: {{ ternary $.Values.grafana.sidecar.dashboards.labelValue "1" (not (empty $.Values.grafana.sidecar.dashboards.labelValue)) | quote }}
+    {{- end }}
+    app: {{ template "kube-prometheus-stack.name" $ }}-grafana
+spec:
+  allowCrossNamespaceImport: true
+  resyncPeriod: {{ .Values.grafana.operator.resyncPeriod | quote | default "10m" }}
+  folder: {{ .Values.grafana.operator.folder | quote }}
+  instanceSelector:
+    matchLabels:
+    {{- with .Values.grafana.operator.matchLabels }}
+      {{- toYaml . | nindent 6 }}
+    {{- end }}
```

Comment:
please double check, which is correct. Normally, matchLabels is omit, if `grafana.operator.matchLabels` would be empty.

```suggestion
  {{- with .Values.grafana.operator.matchLabels }}
  instanceSelector:
    matchLabels:
      {{- toYaml . | nindent 6 }}
  {{- end }}
```

Reply:
actually, this is needed to enforce `grafana-operator` to select the correct instance of `kind: grafana` if not ntothing will appears, that why i've set the `with` like this, but i can change and make a better conditions on it , to enforce a values